src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -0,0 +1,178 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Data
+{
+    public sealed partial class TextLoaderRange
+    {
+        public TextLoaderRange()
+        {
+        }
+
+        /// <summary>
+        /// Convenience constructor for the scalar case.
+        /// Min and Max are set to the single value <paramref name="ordinal"/>.
+        /// When a given column in the schema spans only a single column in the dataset,
+        /// <see cref="ML.Runtime.Data.TextLoader.Range"/>
+        /// </summary>
+        public TextLoaderRange(int ordinal)
+        {
+
+            Contracts.CheckParam(ordinal >= 0, nameof(ordinal));
+
+            Min = ordinal;
+            Max = ordinal;
+        }
+
+        /// <summary>
+        /// Convenience constructor for the vector case.
+        /// When a given column in the schema spans contiguous columns in the dataset,
+        /// <see cref="ML.Runtime.Data.TextLoader.Range"/>
+        /// </summary>
+        public TextLoaderRange(int min, int max)
+        {
+
+            Contracts.CheckParam(min >= 0, nameof(min));
+            Contracts.CheckParam(max >= 0, nameof(max));
```

Comment:
Max should be >= than min. You also do not have any message at all for the user for what actually happened.

The `TextLoader` class is arguably one of the most important components in this API, wouldn't we say? If a user misuses it, and we throw an exception, we really ought to be clear.